File: pset1/mario/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
   int n;

   do
   {
        n = get_int("Height: ");
   }while(n < 1 ||  n > 8);

   int row, a, mid, b, c;
   for ( row = 1; row <= n; row++)
   {
      for (a = n; a > row; a--) //decreasing blank triangle
      {
         printf(" ");
      }
      for (b = 1; b <= row; b++) //increasing leftward triangle
      {
         printf("#");
      }
      for (mid = 1; mid < 2; mid++) //middle space
      {
         printf("  ");
      }
      for (c = 1; c <= row; c++) //increasing rightward triangle
      {
         printf("#");
      }

      printf("\n");
   }
}

// Troubles I had in this problem

// I struggle to make the downward blank decreasing triangle
// The secret: was that I compare the "a" w/ 0 instead of no of rows
//

File: pset6/mario/mario.py
from cs50 import get_int

while True:
    height = get_int("Height: ")
    if height > 0 and height < 9:
        break
for i in range(height):
    for j in range(1, height - i):
        print(" ", end = "")
    for k in range(i + 1):
        print("#", end = "")
    for l in range(2):
        print(" ", end = "")
    for m in range(i + 1):
        print("#", end = "")

    print("")